src/features/ci-cd/hooks/release-versioning.ts

```diff
@@ -29,3 +34,325 @@ hooks.addPureHook('PATCH', 'resin', 'release', {
 		}
 	},
 });
+
+const getAdvisoryLockForApp = async (tx: Tx, appId: number) => {
+	if (!Number.isInteger(appId)) {
+		// This should never happen, since Pine has already validated the value,
+		// but double-check it just to be sure what we are passing to the advisory lock.
+		throw new errors.BadRequestError(
+			'Invalid belongs_to__application parameter',
+		);
+	}
+	await tx.executeSql(`SELECT pg_advisory_xact_lock($1, $2);`, [
+		ADVISORY_LOCK_NAMESPACES.release__revision__belongs_to__application,
+		appId,
+	]);
```

Comment:
I plan to change this to the respective pine db.ts call once that PR gets merged.
See: https://github.com/balena-io/pinejs/pull/495